### PR TITLE
feat: Add support for spaces polyline names

### PR DESCRIPTION
**Pull request by tim-vd-aardweg**
refs: #563

### CODE REVIEW

File: hydrolib/core/dflowfm/polyfile/parser.py [hunk 1]
```diff
@@ -458,7 +458,7 @@ def _is_empty_line(line: str) -> bool:
     @staticmethod
     def _is_name(line: str) -> bool:
         stripped = line.strip()
-        return len(stripped) >= 1 and line[0] != "*" and " " not in stripped
+        return len(stripped) >= 1 and line[0] != "*"
```
Comment (MRVermeulenDeltares) [inline hydrolib/core/dflowfm/polyfile/parser.py:461]: should the documentation of read_polyfile be updated?
It stated:  `- Name consisting of a non-blank character string`

File: hydrolib/core/dflowfm/polyfile/parser.py [hunk 2]
```diff
@@ -458,7 +458,7 @@ def _is_empty_line(line: str) -> bool:
...
 
     @staticmethod
     def _convert_to_name(line: str) -> str:
```
Comment (MRVermeulenDeltares) [inline hydrolib/core/dflowfm/polyfile/parser.py:464]: should `_convert_to_name` still have a .strip() on the name?

File: tests/dflowfm/test_polyfile.py [hunk 3]
```diff
@@ -268,7 +268,7 @@ def test_convert_to_comment(self, input: str, expected_value: str):
             ("*", False),
             ("*         ", False),
             ("", False),
-            ("Not a comment but a regular string", False),
+            ("Not a comment but a regular string", True),
```
Comment (MRVermeulenDeltares) [inline tests/dflowfm/test_polyfile.py:271]: maybe add a case like "Name with spaces".
It is similar to the test Not a comment but a regular string, but it states more clearly what you want to test I think.
You could still keep the Not a comment but a regular string to also indicate a sentence will be seen as a name.

File: tests/dflowfm/test_polyfile.py [hunk 4]
```diff
@@ -595,54 +595,27 @@ def test_empty_lines_is_correctly_logged(
                     """
                     *description
                     name
-                    2  5   
+                    2  5
                     1.0 2.0 3.0 4.0 5.0"""
                 ),
                 [((0, 4), "EoF encountered before the block is finished.")],
             ),
-            (
```
Comment (MRVermeulenDeltares) [inline tests/dflowfm/test_polyfile.py:603]: is it correct to remove all these test cases? ↓

File: hydrolib/core/dflowfm/polyfile/parser.py (hunk 2 above)
Comment (MRVermeulenDeltares) [reply hydrolib/core/dflowfm/polyfile/parser.py:464]: N/A, I misunderstood strip.

File: tests/dflowfm/test_polyfile.py (hunk 4 above)
Comment (MRVermeulenDeltares) [reply tests/dflowfm/test_polyfile.py:603]: Discussed

File: hydrolib/core/dflowfm/polyfile/parser.py (hunk 1 above)
Comment (tim-vd-aardweg) [reply hydrolib/core/dflowfm/polyfile/parser.py:461]: Done

File: tests/dflowfm/test_polyfile.py (hunk 3 above)
Comment (tim-vd-aardweg) [reply tests/dflowfm/test_polyfile.py:271]: Done

**Review by MRVermeulenDeltares — APPROVED** [review]

**Comment by veenstrajelmer** [comment]
Tested and works as expected

**Pull request merged**